src/app/sequence-editor/sequence-editor.component.ts

```diff
@@ -123,4 +126,26 @@ export class SequenceEditorComponent {
     this.filteredCards = this.flashcards.filter(
       item => item.term.toLowerCase().includes(filterValue.toLowerCase()));
   }
```

Comment:
When you search something up and then select the last card in the list, and then you clear the search bar, the arrows get incorrectly displayed and the the buttons move you to the incorrect card. 
#### Initial Search for "le"
![image](https://github.com/PonderTeam/ponder-app/assets/80545567/455e51aa-a058-497a-9f8a-5c32c385296d)
#### Go back to empty search field
![image](https://github.com/PonderTeam/ponder-app/assets/80545567/cf721abe-9c67-4813-8ac3-a1ac77c75277)
#### After pressing the left arrow of from the previous image
![image](https://github.com/PonderTeam/ponder-app/assets/80545567/f1728bb1-ed7a-4b88-ba99-b92452ef36a0)

I believe the problem is because the index you save in the currentCardIndex is not updated when the filtering occurs, so in image 1, the index is set to 2 and then in image 2 the index is still set to 2 when it should be updated to 6. I think you just need to update the currentCardIndex whenever you filterItems.